taskfiles/ci.yml

```diff
@@ -14,11 +14,18 @@ tasks:
       - test -n "$GITHUB_API_TOKEN" || test -n "$GITHUB_TOKEN"
 
   run-kuttl-tests:
+    dir: operator
+    vars:
+      KUTTL_CONFIG_FILE: '{{default "kuttl-test.yaml" .KUTTL_CONFIG_FILE}}'
+      KUTTL_ARTIFACTS_DIR:
+        sh: |
+          cat {{.KUTTL_CONFIG_FILE}} | grep artifactsDir | awk '{print $2}'
     cmds:
       - defer: 'task ci:kuttl-artifact-upload EXIT_CODE={{.EXIT_CODE}} KUTTL_CONFIG_FILE={{.KUTTL_CONFIG_FILE}}'
       - task: :k8s:run-kuttl-tests
         vars:
           KUTTL_CONFIG_FILE: '{{.KUTTL_CONFIG_FILE}}'
+      - buildkite-agent artifact upload "{{.SRC_DIR}}/operator/{{.KUTTL_ARTIFACTS_DIR}}/kuttl-report.xml"
```

Comment:
Is this not already handled by the defer'd task? Should we just make the deferred task always upload artifacts regardless of EXIT_CODE?

Reply:
I configured it to only work when kuttle failed, so that artifacts would not be send to s3.

Reply:
right, wouldn't the fix be to change `ci:kuttl-artifact-upload` to always upload the artifacts instead of adding a duplicate call to artifact upload? Is there a reason why that wouldn't be preferable or possible?

Reply:
My preference is to not bloat S3 with kind logs if test passed. 

I'm fine with doing it in defer and ask devprod to create a cron job to delete 3 weeks old artifacts (maybe less than 3 weeks).